include/grpcpp/security/tls_credentials_options.h

```diff
@@ -0,0 +1,260 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPCPP_SECURITY_TLS_CREDENTIALS_OPTIONS_H
+#define GRPCPP_SECURITY_TLS_CREDENTIALS_OPTIONS_H
+
+#include <memory>
+#include <vector>
+
+#include <grpc/grpc_security.h>
+#include <grpc/support/log.h>
+#include <grpcpp/support/config.h>
+
+namespace grpc_impl {
+namespace experimental {
+
+/** TLS key materials config, wrapper for grpc_tls_key_materials_config. **/
+class TlsKeyMaterialsConfig {
+ public:
+  struct PemKeyCertPair {
+    grpc::string private_key;
+    grpc::string cert_chain;
+  };
+
+  /** Getters for member fields. **/
+  const grpc::string pem_root_certs() const { return pem_root_certs_; }
+  const std::vector<PemKeyCertPair>& pem_key_cert_pair_list() const {
+    return pem_key_cert_pair_list_;
+  }
+  int version() const { return version_; }
+
+  /** Setter for key materials that will be called by the user. The setter
+   * transfers ownership of the arguments to the config. **/
+  void set_key_materials(grpc::string pem_root_certs,
+                         std::vector<PemKeyCertPair> pem_key_cert_pair_list);
+  void set_version(int version) { version_ = version; };
+
+ private:
+  int version_;
+  std::vector<PemKeyCertPair> pem_key_cert_pair_list_;
+  grpc::string pem_root_certs_;
+};
+
+/** The following 2 functions are exposed for testing purposes. **/
+grpc_tls_key_materials_config* c_key_materials(
+    const std::shared_ptr<TlsKeyMaterialsConfig>& config);
+
+std::shared_ptr<TlsKeyMaterialsConfig> tls_key_materials_c_to_cpp(
+    const grpc_tls_key_materials_config* config);
+
+/** TLS credential reload arguments, wraps grpc_tls_credential_reload_arg. **/
+class TlsCredentialReloadArg {
+ public:
+  TlsCredentialReloadArg();
+  TlsCredentialReloadArg(grpc_tls_credential_reload_arg arg);
+  ~TlsCredentialReloadArg();
+
+  /** Getters for member fields. The callback function is not exposed. **/
+  void* cb_user_data() const;
+  std::shared_ptr<TlsKeyMaterialsConfig> key_materials_config() const;
+  grpc_ssl_certificate_config_reload_status status() const;
+  std::shared_ptr<grpc::string> error_details() const;
```

Comment:
Could you add some description specifying these functions will return corresponding fields in C credential reload arg? 
Same for the setters.